starknet_py/net/client_models.py

```diff
@@ -346,6 +346,41 @@ class ContractClass:
     abi: Optional[AbiDictList] = None
 
 
+@dataclass
+class NewEntryPoint:
+    """
+    Dataclass representing contract entry point
+    """
+
+    function_idx: int
+    selector: int
+
+
+@dataclass
+class NewEntryPointsByType:
+    """
+    Dataclass representing contract class entrypoints by entry point type
+    """
+
+    constructor: List[NewEntryPoint]
+    external: List[NewEntryPoint]
+    l1_handler: List[NewEntryPoint]
+
+
+@dataclass
+class NewContractClass:
+    """
+    Dataclass representing Cairo1 contract declared to Starknet
+    """
+
+    contract_class_version: str  # Should this be added? Purpose of it is unclear to me.
+    program: dict
+    entry_points_by_type: NewEntryPointsByType
+    abi: Optional[dict] = None
```

Comment:
I wonder. We made it optional previously because I think it is (or we that it so) possible to have a contract declared without an abi? But not sure about that.

Reply:
Okay. It was made optional because rpc didn't required the `getClass` to return an ABI. For new ContractClass it actually doesn't define if it is required or not, but I'm assuming it will be the same as the DeprecatedContractClass.